### PR TITLE
Better Paste Handling

### PR DESCRIPTION
**Pull request by j0k3r**
Back on June 2015,  I started to work on a different way to handle HTML paste when using Firefox.

The main problem I found when you paste HTML in Firefox it removes every node and paste plain text no matter what paste configuration you have on MediumEditor.

I found that other editor handle that problem by cheating on the way Firefox handle contenteditable. When you paste HTML in a contenteditable, Firefox paste plain text only.

So the way to handle HTML paste in Firefox is:
- catch the paste event
- move cursor in a non `contenteditable` node (like an hidden `div` in the page) BEFORE the even is finalized
- the even is dispatched and the html content is in the non `contenteditable` node
- move this content to the MediumEditor node

Since the bug is still here (see https://github.com/yabwe/medium-editor/issues/972) I push my work on that part.

As far as I remember the first paste worked but all other one failed ...

### CODE REVIEW

**Comment by Dayjo** [comment]
Will this affect the usage of pasting from the context or edit menus (a surprising number of people do it this way).

**Comment by j0k3r** [comment]
This first draft only handle the <kbd>CTRL</kbd>+<kbd>B</kbd> but yeah, we'll need to handle context menu

File: src/js/extensions/paste.js [hunk 1]
```diff
@@ -86,48 +89,128 @@
...
+
+                // Grab plain text from Clipboard API or convert existing HTML to plain text
+                if (plainTextMode) {
+                    // Use plain text contents from Clipboard API unless the HTML contains paragraphs then
+                    // we should convert the HTML to plain text since works better when pasting HTML/Word contents as plain text
+                    if (that.hasContentType(clipboardContent, 'text/plain') && content.indexOf('</p>') === -1) {
+                        content = clipboardContent['text/plain'];
+                    } else {
+                        // What is innerText supposed to do? (It isn't defined)
+                        //content = MediumEditor.util.innerText(content);
```
Comment (j0k3r) [inline src/js/extensions/paste.js]: Hum good point, I'll have a look on monday

Comment (j0k3r) [inline src/js/extensions/paste.js]: I found nothing about that on my work dev machine. I guess it was to retrieve the html from `content`? I can't remember

**Comment by j0k3r** [comment]
Huge thanks @nmielnik for finishing this stuff 👍 🚀

**Pull request merged**